### ISSUE TITLE
[0.57.2] run_target with a missing quoted command started requiring the binary to exists

### ISSUE THREAD

**Issue by keszybz** [mesonbuild/meson#8641]
**Describe the bug**

We have the following code in systemd:
```
run_target(
        'doc-sync',
        depends : man_pages + html_pages,
        command : ['rsync', '-rlv', ...])
```

This target is only used when manually invoked. So far, the presence of `rsync` was not probed, so it was OK if `rsync` was not installed (until the target was executed). This changed in 0.57.2 (0.57.1 was still OK). So for example CI builds started failing. Please consider reverting the change, it doesn't seem to be something suitable for a point release.

```
project('a', 'c',
        version : '0')

run_target('run-foobar',
           command : ['foobar', '!'])
```

With meson-0.57.2-1.fc35.noarch, this fails.

**Comment by keszybz** [comment]
Oh, this has another impact:
```
run_target(
        'fuzzers',
        depends : fuzzer_exes,
        command : ['true'])
```
(The "true" is a placeholder, required because run_target requires a command to be specified.)

This now started printing:
```
Program true found: YES (/usr/bin/true)
```

**Comment by xclaesse** [comment]
This is probably caused by https://github.com/mesonbuild/meson/pull/8526.

I think it's the right thing to do, but I agree it probably should not be in a point release. That change was needed to fix another bug, and we probably overlooked the fact that the program was not required before.

If your target is optional, do something like:
```meson
rsync = find_program('rsync', required: false)
if rsync.found()
  run_target(..., command: [rsync, ...])
endif
```

**Comment by keszybz** [comment]
> ```meson
> rsync = find_program('rsync', required: false)
> if rsync.found()
>   run_target(..., command: [rsync, ...])
> endif
> ```

Yep, https://github.com/systemd/systemd/pull/19297 does exactly this.

**Comment by eli-schwartz** [comment]
find_program() supports `disabler: true`

**Comment by keszybz** [comment]
> find_program() supports `disabler: true`

Oh, you mean that we could do
```python
rsync = find_program('rsync', required: false)
run_target(..., command: [rsync, ...])
```

In systemd we still support meson-0.46 sadly…

**Comment by eli-schwartz** [comment]
Oh well, leave it as a `# TODO`. :)

**Comment by bonzini** [comment]
Based on [this comment](https://github.com/mesonbuild/meson/issues/8641#issuecomment-817802997) this issue can be closed as intended behavior.
